app/components/markdown/markdown_latex_block/index.tsx

```diff
@@ -88,14 +89,33 @@ const LatexCodeBlock = ({content, theme}: Props) => {
     const insets = useSafeAreaInsets();
     const managedConfig = useManagedConfig<ManagedConfig>();
     const styles = getStyleSheet(theme);
+    const languageDisplayName = getHighlightLanguageName('latex');
+
+    const splitContent = () => {
+        const lines = splitLatexCodeInLines(content);
+        const numberOfLines = lines.length;
+
+        if (numberOfLines > MAX_LINES) {
+            return {
+                content: lines.slice(0, MAX_LINES),
+                numberOfLines,
+            };
+        }
 
-    const handlePress = preventDoubleTap(() => {
+        return {
+            lines,
+            numberOfLines,
+        };
+    };
+
+    const prevContent = useRef(content);
+    const [split, setSplit] = useState(splitContent());
```

Comment:
Not 100% sure about this due to React magic, but this seems to execute `splitContent()` on every render, to pass the value to the function, even if useState is not going to use it at all.

Also, seems like setSplit only is called in the useEffect. That behaviour is practically the same as using here a useMemo depending on the same values as the effect, and with one render less (after the effect, since the state changed, there is another render). Unless this double render is needed, I would consider making this a memo.

Reply:
good point